tensorflow/python/keras/utils/vis_utils.py

```diff
@@ -124,6 +147,26 @@ def model_to_dot(model,
     dot.set('dpi', dpi)
     dot.set_node_defaults(shape='record')
 
+  if layer_range is None:
+    layer_range = [0, len(model.layers)]
+  
+  # check point for layer range whether its string or integer 
+  # layer range must be same dtype (mixed dtype is not allowed)
+  if all(map(lambda x: str(x).isdigit(), layer_range)):
+    layer_range = list(map(int, layer_range))
+    if len(layer_range) >= 2:
+      layer_range = [min(layer_range), max(layer_range)]
```

Comment:
Not sure why we do this here. If the input is [1, 2, 4], shouldn't we just raise error here?

Reply:
Passing more than two parameters for `layer range` makes less sense but instead of raising an error, it takes the lower and upper bound to plot the model anyway but raising an error probably makes more sense and can be done also.

Reply:
passing more than 2 elements doesn't make any sense. Let's say if we pass `[1, 2, 4]`, then we are not utilizing `2` at any point so there is no point giving flexibility to user for passing more than 2 elements.